wurst/objects/abilities/TrollUpgradeSpell.wurst

```diff
@@ -20,86 +19,217 @@ import TrollUnitTextConstant
-
-        // Beast Master
-        new TrollUpgradeSubSpell(ABILITY_SHAPESHIFTER         , "Shapeshifter"         , TOOLTIP_SHAPESHIFTER    , new Pair(1, 0), "BearForm"                , "Q"),
-        new TrollUpgradeSubSpell(ABILITY_CHICKEN              , "Chicken"              , TOOLTIP_CHICKEN         , new Pair(2, 0), "CritterChicken"          , "W"),
-        new TrollUpgradeSuperSubSpell(ABILITY_ULTIMATE_FORM   , "Ultimate Form"        , TOOLTIP_ULTIMATE_FORM   , new Pair(3, 0), "JungleBeast"             , "E"),
-
-        // Thief
-        new TrollUpgradeSubSpell(ABILITY_ESCAPE_ARTIST        , "Escape Artist"        , TOOLTIP_ESCAPE_ARTIST   , new Pair(1, 0), "DarkTroll"               , "Q"),
-        new TrollUpgradeSubSpell(ABILITY_CONTORTIONIST        , "Contortionist"        , TOOLTIP_CONTORTIONIST   , new Pair(2, 0), "Contortionist"           , "W"),
-        new TrollUpgradeSuperSubSpell(ABILITY_ASSASSIN        , "Assassin"             , TOOLTIP_ASSASSIN        , new Pair(3, 0), "DarkTrollTrapper"        , "E"),
-
-        // Scout
-        new TrollUpgradeSubSpell(ABILITY_OBSERVER             , "Observer"             , TOOLTIP_OBSERVER        , new Pair(1, 0), "ObserverSentry"          , "Q"),
-        new TrollUpgradeSubSpell(ABILITY_TRAPPER              , "Trapper"              , TOOLTIP_TRAPPER         , new Pair(2, 0), "Trapper"                 , "W"),
-        new TrollUpgradeSuperSubSpell(ABILITY_SPY             , "Spy"                  , TOOLTIP_SPY             , new Pair(3, 0), "ForestTrollTrapper"      , "E"),
-
-        // Gatherer
-        new TrollUpgradeSubSpell(ABILITY_HERB_MASTER          , "Herb Master Gatherer" , TOOLTIP_HERB_MASTER     , new Pair(1, 0), "IceTrollShadowPriest"    , "Q"),
-        new TrollUpgradeSubSpell(ABILITY_RADAR_GATHERER       , "Radar Gatherer"       , TOOLTIP_RADAR_GATHERER  , new Pair(2, 0), "TerrorTroll"             , "W"),
-        new TrollUpgradeSuperSubSpell(ABILITY_OMNIGATHERER    , "Omnigatherer"         , TOOLTIP_OMNIGATHERER    , new Pair(3, 0), "TerrorTroll"             , "E")
-    )
-
-@compiletime function buildSpell()
-    getTrollUpgrade().forEach(elem -> elem.buildSpell())
+
+@compiletime function createUpgradeSpellWarrior() returns ChannelAbilityPreset
+    return createUpgradeSpell(ABILITY_WARRIOR)
+        ..setUpgradeSubSpell("Warrior", TOOLTIP_WARRIOR)
+        ..setIconNormal(LocalIcons.bTNTrollPredator)
+        ..setIconResearch(LocalIcons.bTNTrollPredator)
+        ..setRequiredLevel(6)
+        ..presetHotkey("Q")
+        ..presetButtonPosResearch(1, 0)
+
+@compiletime function createUpgradeSpellTracker() returns ChannelAbilityPreset
+    return createUpgradeSpell(ABILITY_TRACKER)
+        ..setUpgradeSubSpell("Tracker", TOOLTIP_TRACKER)
+        ..setIconNormal(Icons.bTNIceTroll)
+        ..setIconResearch(Icons.bTNIceTroll)
+        ..setRequiredLevel(6)
+        ..presetHotkey("W")
+        ..presetButtonPosResearch(2, 0)
```

Comment:
Do these research fields matter?

Reply:
They did when the spells were chosen in hero skill menu, subclass menu is a spell book right? In that case, it's useless now.